app/models/user.rb

```diff
@@ -73,7 +73,11 @@ def current_roles
   end
 
   def preferred_language
-    language = profile.language
+    language = nil
+    unless profile.blank?
```

Comment:
Instead of instantiating the variable nil, then changing it according to profile, I'd do a ternary.

`language = profile.blank? ? nil : profile.language`